pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -212,6 +212,7 @@ type CertificatePrivateKey struct {
 	// will be generated whenever a re-issuance occurs.
 	// Default is 'Never' for backward compatibility.
 	// +optional
+	// +kubebuilder:validation:Enum=Never;Always
```

Comment:
Great. 

I was about to suggest you also add this validation annotation to the legacy API versions (which still exist), but then remembered that we're not generating CRDs for those versions, so ignore this comment: 
 https://github.com/cert-manager/cert-manager/blob/1525026c999e8e1dbea367c8b2102ae7f8d86a6f/internal/apis/certmanager/v1beta1/types_certificate.go#L197-L211 


We kept those legacy versions for use by `cmctl convert`, but I expect that if it tried to convert a v1beta1 API manifest which had an invalid RotationPolicy value, that that value would simply be copied to the v1 YAML and then this validation would alert the user when they tried to apply it.